### DISCUSSION TITLE
Space character underlined in green

### DISCUSSION THREAD

**Discussion by JPhB** [vkbo/novelWriter#1347] [Help] [answered]
I like novelWriter.

When I type a word, and then a space, the space is underlined in green. As soon as I continue typing the underline disappears. 

![Capture](https://user-images.githubusercontent.com/39348213/216843351-f3719666-c768-4f2c-89fa-1603da99b10d.JPG)

Why is that? 

I also would like to know if there is a way to automatically capitalize first letters of sentences.

**Comment by vkbo** [comment, marked as answer]
> When I type a word, and then a space, the space is underlined in green. As soon as I continue typing the underline disappears. Why is that?

There is a feature that underlines redundant spaces, and a space at the end of a line is considered redundant. As soon as you type something more, it is no longer redundant and the highlighing goes away.

You can turn it off in "Preferences > Highlighting > Highlight multiple or trailing spaces" if you wish.

> I also would like to know if there is a way to automatically capitalize first letters of sentences.

No, there is no such feature at the moment. I've thought about adding it though.

**Reply by zeroanaphora** [reply]
Honestly I'm surprised this is default behavior, it flickering on and off as you type is incredibly distracting. Glad I found this comment.

**Reply by vkbo** [reply]
I don't notice it, so it's never bother me, but with some testing I see that changing the highlighting rule from `[ ]{2,}|[ ]*$` to `[ ]{2,}|(?: \n)` should delay the trailing space highlight until you press enter. This would stop the temporary highlight when just typing spaces between words.

**Reply by vkbo** [reply]
Created ticket #2501, so this should be fixed in the next release, version 2.8.
